### DISCUSSION TITLE
Search in branches

### DISCUSSION THREAD

**Discussion by 111andre111** [community/community#16972] [Other Feature Feedback, Questions, & Ideas]
Hi folks,

I would like to have in Github Code Search the possibility to search inside branches.

so a search item like

`branch:featurebranch` would be extremely cool.

Is there any chance to get a feature like that?

**Comment by look** [comment]
Hi @111andre111, this is definitely something on our radar. It's difficult at GitHub's scale so we are doing some fundamental work first. There was [a previous discussion about this](https://github.com/github/feedback/discussions/8564) if you'd like to take a look.

**Comment by ghost** [comment]
@111andre111 Hi! I thought of something similar to what was proposed here: [Search in branches](https://github.com/github/feedback/discussions/16972) and here [Add branch as a search filter](https://github.com/github/feedback/discussions/8564). Note: The difference between these 2 features for my feature I request is that you can control the branch, and after controlling the branch - you could search the public or private branch - reference [here](https://github.com/github/feedback/discussions/16334).

**Comment by Neustradamus** [comment]
At this time, in 2025, I have always the problem, the search is only with default branch.

It is possible to add a solution?

Thanks in advance.
